maitake/src/wait/cell.rs

```diff
@@ -237,8 +303,37 @@ impl fmt::Debug for State {
     }
 }
 
+#[cfg(all(feature = "alloc", not(loom), test))]
+mod test {
+    use super::*;
+    use crate::scheduler::Scheduler;
+    use alloc::sync::Arc;
+
+    #[test]
+    fn wait_smoke() {
+        static COMPLETED: AtomicUsize = AtomicUsize::new(0);
+
+        let sched = Scheduler::new();
+        let wait = Arc::new(WaitCell::new());
+
+        let wait2 = wait.clone();
+        sched.spawn(async move {
+            wait2.wait().await.unwrap();
+            COMPLETED.fetch_add(1, Ordering::Relaxed);
+        });
+
+        let tick = sched.tick();
+        assert_eq!(tick.completed, 0);
+        assert_eq!(COMPLETED.load(Ordering::Relaxed), 0);
+
+        assert_eq!(wait.wake(), true);
+        let tick = sched.tick();
+        assert_eq!(tick.completed, 1);
+        assert_eq!(COMPLETED.load(Ordering::Relaxed), 1);
+    }
+}
```

Comment:
also, it occurs to me that since this is testing only `WaitCell`'s public API, it _might_ belong in `tests/` as an integration test, rather than in `wait::cell::tests` as a unit test...but...i don't actually care...